internal/os/volume/api.go

```diff
@@ -41,9 +67,9 @@ func getVolumeSize(volumeID string) (int64, error) {
 	return volumeSize, nil
 }
 
-// ListVolumesOnDisk - returns back list of volumes(volumeIDs) in the disk (requested in diskID).
-func (VolAPIImplementor) ListVolumesOnDisk(diskID string) (volumeIDs []string, err error) {
-	cmd := fmt.Sprintf("(Get-Disk -DeviceId %s |Get-Partition | Get-Volume).UniqueId", diskID)
+// ListVolumesOnDisk - returns back list of volumes(volumeIDs) in the disk.
+func (VolumeAPI) ListVolumesOnDisk(diskNumber uint32) (volumeIDs []string, err error) {
```

Comment:
partition number is added as a second parameter

Reply:
Get-Partition
   [[-DiskNumber] <UInt32[]>]
   [[-PartitionNumber] <UInt32[]>]
   [-CimSession <CimSession[]>]
   [-ThrottleLimit <Int32>]
   [-AsJob]
   [<CommonParameters>]

So I think you can use Get-Partition -DiskNumber %d -PartitionNumber %d

Reply:
correct, I forgot to ask about using this parameter which is 0 by default, thanks for noticing this!

Reply:
do you think that this is ok? The default value set by gRPC is [0 for unsigned int](https://developers.google.com/protocol-buffers/docs/proto3#default) so the command is like this:

```
// ListVolumesOnDisk - returns back list of volumes(volumeIDs) in a disk and a partition.
func (VolumeAPI) ListVolumesOnDisk(diskNumber uint32, partitionNumber uint32) (volumeIDs []string, err error) {
	// partitionNumber=0 means that the partitionNumber wasn't set so we list all the partitions
	var cmd string
	if partitionNumber == 0 {
		cmd = fmt.Sprintf("(Get-Disk -Number %d | Get-Partition | Get-Volume).UniqueId", diskNumber)
	} else {
		cmd = fmt.Sprintf("(Get-Disk -Number %d | Get-Partition -PartitionNumber %d | Get-Volume).UniqueId", diskNumber, partitionNumber)
	}
```

Reply:
it looks fine.